src/parse/text/simple_extensions/argument.rs

```diff
@@ -0,0 +1,633 @@
+        if options.is_empty() {
+            return Err(EnumOptionsError::EmptyList);
+        }
+
+        let mut unique_set = std::collections::HashSet::new();
+        for option in options.iter() {
+            if option.is_empty() {
+                return Err(EnumOptionsError::EmptyOption);
+            }
+            if !unique_set.insert(option) {
+                return Err(EnumOptionsError::DuplicatedOption(option.clone()));
+            }
+        }
+
+        Ok(EnumOptions(options))
+    }
+}
+
+impl From<EnumOptions> for simple_extensions::EnumOptions {
+    fn from(value: EnumOptions) -> Self {
+        simple_extensions::EnumOptions(value.0)
+    }
+}
+
+/// Parse errors for [simple_extensions::EnumOptions].
+#[derive(Debug, Error, PartialEq)]
+pub enum EnumOptionsError {
+    /// Empty list.
+    #[error("empty list")]
+    EmptyList,
+
+    /// Duplicated option.
+    #[error("duplicated option: {0}")]
+    DuplicatedOption(String),
+
+    /// Empty option.
+    #[error("empty option")]
+    EmptyOption,
+}
+
+/// Arguments that refer to a data value.
+#[derive(Clone, Debug)]
+pub struct ValueArg {
+    /// A human-readable name for this argument to help clarify use.
+    name: Option<String>,
+
+    /// Additional description for this argument.
+    description: Option<String>,
+
+    /// A fully defined type or a type expression.
+    ///
+    /// todo: implement parsed [simple_extensions::Type].
+    value: simple_extensions::Type,
+
+    /// Whether this argument is required to be a constant for invocation.
+    /// For example, in some system a regular expression pattern would only be accepted as a literal
+    /// and not a column value reference.
+    constant: Option<bool>,
+}
+
+impl<C: Context> Parse<C> for simple_extensions::ValueArg {
+    type Parsed = ValueArg;
+
+    type Error = ArgumentsItemError;
+
+    fn parse(self, _ctx: &mut C) -> Result<ValueArg, ArgumentsItemError> {
+        Ok(ValueArg {
+            name: ArgumentsItem::parse_name(self.name)?,
+            description: ArgumentsItem::parse_description(self.description)?,
+            value: self.value,
+            constant: self.constant,
+        })
+    }
+}
+
+impl From<ValueArg> for simple_extensions::ValueArg {
+    fn from(value: ValueArg) -> Self {
+        simple_extensions::ValueArg {
+            name: value.name,
+            description: value.description,
+            value: value.value,
+            constant: value.constant,
+        }
+    }
+}
+
+impl From<ValueArg> for simple_extensions::ArgumentsItem {
+    fn from(value: ValueArg) -> Self {
+        simple_extensions::ArgumentsItem::ValueArg(value.into())
+    }
+}
+
+impl From<ValueArg> for ArgumentsItem {
+    fn from(value: ValueArg) -> Self {
+        ArgumentsItem::ValueArgument(value)
+    }
+}
+
+/// Arguments that are used only to inform the evaluation and/or type derivation of the function.
+#[derive(Clone, Debug, PartialEq)]
+pub struct TypeArg {
```

Comment:
And can you also add methods for this one?

Reply:
Added in [commit](https://github.com/substrait-io/substrait-rs/pull/185/commits/53a2a22eb4e76716ac5b016ae46274a3e5524479).